### PR TITLE
Add TriggerCRD object validation and default

### PR DESCRIPTION
**Pull request by khrm**
Defaults and Validation for TriggerCRD object have been added

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

**Review by savitaashture — COMMENTED** [review]
I see code duplication for `defaults` and `validator` between eventlistener and triggers 

Can we avoid code duplication by writing common function and calling from both the place?

File: pkg/apis/triggers/v1alpha1/trigger_defaults.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+// set default TriggerBinding kind for Bindings
+func (t *Trigger) defaultBindings() {
+	if len(t.Spec.Bindings) > 0 {
+		for _, b := range t.Spec.Bindings {
+			if b.Kind == "" {
+				b.Kind = NamespacedTriggerBindingKind
+			}
+		}
+	}
```
[the remark this answers is not in the document]
Comment (khrm) [reply pkg/apis/triggers/v1alpha1/trigger_defaults.go:39]: Problem is that types are different. Maybe I need a create a pr which makes type same in the same way we do for Interceptor.

[the remark this answers is not in the document]
Comment (khrm) [reply pkg/apis/triggers/v1alpha1/trigger_defaults.go:39]: I raise a pr which does that here: https://github.com/tektoncd/triggers/pull/740

If that is merged, I would change the code to be similar.

**Comment by dibyom** [comment]
I'd mark this as `kind/feature` since we are adding something that a user can use i.e extra validation/defaults.

File: pkg/apis/triggers/v1alpha1/trigger_validation_test.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+	tests := []struct {
+		name string
+		tr   *v1alpha1.Trigger
+	}{{
+		name: "Valid Trigger No TriggerBinding",
+		tr: bldr.Trigger("name", "namespace",
+			bldr.TriggerSpec(
+				bldr.TriggerSpecTemplate("tt", "v1alpha1"))),
+	}, {
+		name: "Valid EventListener with TriggerBinding",
```
Comment (savitaashture) [inline pkg/apis/triggers/v1alpha1/trigger_validation_test.go:39]:
```suggestion
		name: "Valid Trigger with TriggerBinding",
```

**Pull request merged**